Automattic/simplenote-ios
# Nü passcode screen: IV

### Fix
In this PR we're finally removing `DTPinLockController` and replacing it with `PinLockViewController`. The final piece is the passcode screen to unlock the app.
The app should behave exactly as it was, but with new shiny passcode UI.

Related issue: #761

|Before|After|
|-|-|
|![Simulator Screen Shot - iPhone 12 - 2021-01-05 at 21 24 18](https://user-images.githubusercontent.com/54751/103695205-761c7100-4f9c-11eb-9ef5-36145abb5d4f.png)|![Simulator Screen Shot - iPhone 12 - 2021-01-05 at 21 17 59](https://user-images.githubusercontent.com/54751/103695218-79aff800-4f9c-11eb-9592-ff51de785a54.png)|
|![Simulator Screen Shot - iPad Pro (9 7-inch) - 2021-01-05 at 21 25 42](https://user-images.githubusercontent.com/54751/103695291-96e4c680-4f9c-11eb-9465-6ad84e7740bf.png)|![Simulator Screen Shot - iPad Pro (9 7-inch) - 2021-01-05 at 21 18 08](https://user-images.githubusercontent.com/54751/103695294-977d5d00-4f9c-11eb-8703-fd10741c129f.png)|

### Test (Passcode)
1. Set a passcode
2. Move the app to the background
3. Open multitasking view
4. Check that passcode screen is visible and covers the whole screen
5. Open the app

- [x] Check that there is no "Cancel" button and it is not possible to dismiss the screen
- [x] Entering incorrect passcode shakes the header and shows an error message with the number of attempts
- [x] Entering correct passcode dismisses the screen and gives access to the app

### Test (Biometry)
1. Set a passcode
2. Enable Touch ID / Face ID
3. Move the app to the background
4. Open the app

- [x] If it's the first time using Touch ID / Face ID, system permission dialog will be shown
- [x] Using Touch ID / Face ID let you unlock the app
- [x] Providing wrong finger / face let you cancel Touch ID / Face ID flow and enter a passcode

### Test (Timeout)
1. Set a passcode
2. Set a timeout
3. Move the app to the background
4. Open the app

- [x] Opening the app before timeout unlocks the app without the need to enter passcode
- [x] Opening the app after timeout blocks the app with passcode screen

### Review
Only one developer and one designer are required to review these changes, but anyone can perform the review.

### Release

`RELEASE-NOTES.txt` was updated in 37d767e4 with:
 
> New passcode interface
